scripts/components/App.js

```diff
@@ -42,32 +96,33 @@ export default class App extends React.Component {
 
   constructor(props) {
     super(props);
-    this.state = this.props.store.state;
-    this.props.store.on("change", state => {
-      this.setState(Object.assign({busy: false}, state));
-    });
-    this.props.store.on("error", error => {
-      this.setState({busy: false, error: error.message});
-    });
-    this.props.store.load();
+    this.state = {busy: false, error: "", items: []};
   }
 
-  updateRecord(record) {
-    this.props.store.create(record);
+  componentDidMount() {
+    this.props.events.on("store:busy", state => {
+      this.setState({busy: state, error: ""});
+    });
+    this.props.events.on("store:change", state => {
+      this.setState({items: state.items});
+    });
+    this.props.events.on("store:error", error => {
+      this.setState({error: error.message});
+    });
   }
 
-  syncRecords() {
-    this.setState({busy: true, error: ""});
-    this.props.store.sync();
+  onSync() {
```

Comment:
Nit: `onSyncClick`; it's usually useful to have the event type part of the handling method name.